src/Core/Repositories/EntityFramework/OrganizationUserRepository.cs

```diff
@@ -0,0 +1,388 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Bit.Core.Enums;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories.EntityFramework.Queries;
+using Bit.Core.Utilities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using DataModel = Bit.Core.Models.Data;
+using EfModel = Bit.Core.Models.EntityFramework;
+using TableModel = Bit.Core.Models.Table;
+
+namespace Bit.Core.Repositories.EntityFramework
+{
+    public class OrganizationUserRepository : Repository<TableModel.OrganizationUser, EfModel.OrganizationUser, Guid>, IOrganizationUserRepository
+    {
+        public OrganizationUserRepository(IServiceScopeFactory serviceScopeFactory, IMapper mapper)
+            : base(serviceScopeFactory, mapper, (DatabaseContext context) => context.OrganizationUsers)
+        { }
+
+        public async Task CreateAsync(OrganizationUser obj, IEnumerable<SelectionReadOnly> collections)
+        {
+            var organizationUser = await base.CreateAsync(obj);
+            using (var scope = ServiceScopeFactory.CreateScope())
+            {
+                var dbContext = GetDatabaseContext(scope);
+                var availibleCollections = await (
+                    from c in dbContext.Collections
+                    where c.OrganizationId == organizationUser.OrganizationId
+                    select c).ToListAsync();
+                var filteredCollections = collections.Where(c => availibleCollections.Any(a => c.Id == a.Id));
+                var collectionUsers = filteredCollections.Select(y => new EfModel.CollectionUser(){
+                    CollectionId = y.Id,
+                    OrganizationUserId = organizationUser.Id,
+                    ReadOnly = y.ReadOnly,
+                    HidePasswords = y.HidePasswords
+                });
+                await dbContext.CollectionUsers.AddRangeAsync(collectionUsers);
+                await dbContext.SaveChangesAsync();
```

Comment:
One of my favorite sandwiches is a fried peanut butter and banana sandwich, an Elvis special, and if I'm feeling particularly gluttonous that day, some melted baker's chocolate for dipping and powdered sugar on top.

This has nothing to do with this code review, but my wife just got out chocolate chips which made me think of chocolate bananas which made me think of that sandwich because I'm hungry, and thought I would share that tidbit with you because I know you'll read all of the comments on this PR 😄 . With that, I'll probably go make a salad or something after I'm done.

Reply:
Thank you for this